packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts

```diff
@@ -344,36 +320,25 @@ export default class ImageEdit implements EditorPlugin {
      */
     private createWrapper(operation: ImageEditOperation | CompatibleImageEditOperation) {
         //Clone the image and insert the clone in a entity
-        const clone = this.image.cloneNode();
-        const wrappedImage = wrap(clone, KnownCreateElementDataIndex.ImageEditWrapper);
-
-        this.clonedImage = wrappedImage.firstElementChild as HTMLImageElement;
-
-        // Wrap the cloned image with an entity so that we can easily retrieve it later
-        const { wrapper } = insertEntity(
-            this.editor,
-            IMAGE_EDIT_WRAPPER_ENTITY_TYPE,
-            wrappedImage,
-            false /*isBlock*/,
-            true /*isReadonly*/
-        );
-
-        wrapper.id = this.imageId;
-        wrapper.style.maxWidth = '100%';
-        wrapper.style.position = 'fixed';
+        this.clonedImage = this.image.cloneNode(true) as HTMLImageElement;
+        const wrappedImage = wrap(this.clonedImage, KnownCreateElementDataIndex.ImageEditWrapper);
+        this.wrapper = this.editor.getDocument().createElement('span');
+        this.wrapper.appendChild(wrappedImage);
+        this.wrapper.style.maxWidth = '100%';
+        this.wrapper.style.position = 'fixed';
         // keep the same vertical align
         const originalVerticalAlign = this.getStylePropertyValue(this.image, 'vertical-align');
         if (originalVerticalAlign) {
-            wrapper.style.verticalAlign = originalVerticalAlign;
+            this.wrapper.style.verticalAlign = originalVerticalAlign;
         }
 
-        wrapper.style.display = Browser.isSafari ? 'inline-block' : 'inline-flex';
+        this.wrapper.style.display = Browser.isSafari ? 'inline-block' : 'inline-flex';
 
         // Cache current src so that we can compare it after edit see if src is changed
         this.lastSrc = this.image.getAttribute('src');
 
         // Set image src to original src to help show editing UI, also it will be used when regenerate image dataURL after editing
-        this.image.style.visibility = 'hidden';
+        this.image.className = STYLE_IMAGE;
```

Comment:
Actually the image should already have id. If not, you add some API in editor to ensure it. So no need to change its className